scripts/lighthouse/index.mjs

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const stoplight = res => (res >= 90 ? '🟢' : res >= 75 ? '🟠' : '🔴');
+const normalizeScore = res => Math.round(res * 100);
+const formatScore = res => {
+  const normalizedScore = normalizeScore(res);
+  return `${stoplight(normalizedScore)} ${normalizedScore}`;
+};
+
+/**
+ * `core` is in scope from https://github.com/actions/github-script
+ */
+export const formatLighthouseResults = ({ core }) => {
+  // this will be the shape of https://github.com/treosh/lighthouse-ci-action#manifest
+  const results = JSON.parse(process.env.LIGHTHOUSE_RESULT);
+
+  // this will be the shape of https://github.com/treosh/lighthouse-ci-action#links
+  const links = JSON.parse(process.env.LIGHTHOUSE_LINKS);
+
+  // start creating our markdown table
+  const header = [
+    'Lighthouse Results',
+    'URL | Performance | Accessibility | Best Practices | SEO | Report',
+    '| - | - | - | - | - | - |',
+  ];
+
+  // map over each url result, formatting and linking to the output
+  const urlResults = results.map(({ url, summary }) => {
+    // make the tested link as a markdown link, without the long-generated host
+    const shortPreviewLink = `[${url.replace(
+      process.env.VERCEL_PREVIEW_URL,
```

Comment:
Can you assign this to a constant and have a fallback URL?

Reply:
same as the other comment - an unset variable here will be immediately noticable as something amiss in the workflow or update action supplying the output - prompting us to investigate. I see diminishing returns on a "graceful degratation" of this scipted code when it's for such a tertiary element of our codebase/ci/cd pipeline.

Reply:
That's not what I meant. If this runs against prod branches there's no VERCEL_PREVIEW_URL

Reply:
I'm still learning actions syntax. Doesn't https://github.com/nodejs/nodejs.org/pull/6047/files#diff-46310a6112abe0634d5021fbaff24cd06630a5bad4be4c785cd482b24a0b818fR12-R16 mean that it only runs on pull requests ====_to_==== `main` ?

Reply:
True, right only runs on preview branches.